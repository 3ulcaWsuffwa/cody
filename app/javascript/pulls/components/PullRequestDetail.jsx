// @flow

import React from "react";
import Reviewer from "./Reviewer";
import { createFragmentContainer, graphql } from "react-relay";
import { type PullRequestDetail_pullRequest } from "./__generated__/PullRequestDetail_pullRequest.graphql";

const PullRequestDetail = ({
  pullRequest
}: {
  pullRequest: PullRequestDetail_pullRequest
}) => {
  return (
    <section className="section">
      <div className="container box code">
        <div className="pa4 br2 ba b--light-gray mt4 bg-white">
          <h1 className="title code">
            {`${pullRequest.repository}#${pullRequest.number}`}
          </h1>
          <h2 className="subtitle code">
            {pullRequest.status}
          </h2>
        </div>
        <hr />
        <div>
          {pullRequest.reviewers.edges.map(edge => {
            return <Reviewer key={edge.node.id} reviewer={edge.node} />;
          })}
        </div>
      </div>
    </section>
  );
};

export default createFragmentContainer(
  PullRequestDetail,
  graphql`
    fragment PullRequestDetail_pullRequest on PullRequest {
      id
      repository
      number
      status
      reviewers {
        edges {
          node {
            ...Reviewer_reviewer
          }
        }
      }
    }
  `
);
