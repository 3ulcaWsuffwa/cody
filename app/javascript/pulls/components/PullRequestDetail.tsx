import React from "react";
import Reviewer from "./Reviewer";
import { createFragmentContainer, graphql } from "react-relay";
import { PullRequestDetail_pullRequest } from "./__generated__/PullRequestDetail_pullRequest.graphql";

const PullRequestDetail = ({
  pullRequest
}: {
  pullRequest: PullRequestDetail_pullRequest
}) => {
  return (
    <section className="section">
      <div className="container box code">
        <div className="pa4 br2 ba b--light-gray mt4 bg-white">
          <h1 className="title code">
            {`${pullRequest.repository}#${pullRequest.number}`}
          </h1>
          <h2 className="subtitle code">{pullRequest.status}</h2>
        </div>
        <hr />
        <div>
          {pullRequest.reviewers != null && pullRequest.reviewers.edges != null
            ? pullRequest.reviewers.edges.map(edge => {
                if (edge != null && edge.node != null) {
                  // https://github.com/facebook/relay/issues/1918
                  return <Reviewer key={edge.node.id} reviewer={edge.node} />;
                }
              })
            : null}
        </div>
      </div>
    </section>
  );
};

export default createFragmentContainer(
  PullRequestDetail,
  {
    pullRequest: graphql`
      fragment PullRequestDetail_pullRequest on PullRequest {
        id
        repository
        number
        status
        reviewers {
          edges {
            node {
              id
              ...Reviewer_reviewer
            }
          }
        }
      }
    `
  }
);
