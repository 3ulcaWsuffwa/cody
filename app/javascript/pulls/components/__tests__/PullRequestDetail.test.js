import React from "react";
import renderer from "react-test-renderer";
import PullRequestDetail from "../PullRequestDetail";

test("PullRequestDetail snapshot test", () => {
  const props = {
    repository: "aergonaut/testrepo",
    number: "1234",
    status: "pending_review"
  };
  const component = renderer.create(<PullRequestDetail pullRequest={props} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
