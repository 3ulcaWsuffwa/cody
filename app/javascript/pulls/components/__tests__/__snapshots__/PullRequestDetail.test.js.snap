// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PullRequestDetail snapshot test 1`] = `
<DocumentFragment>
  <section
    class="section"
  >
    <div
      class="container box code"
    >
      <div
        class="pa4 br2 ba b--light-gray mt4 bg-white"
      >
        <h1
          class="title code"
        >
          &lt;mock-value-for-field-"repository"&gt;#&lt;mock-value-for-field-"number"&gt;
        </h1>
        <h2
          class="subtitle code"
        >
          &lt;mock-value-for-field-"status"&gt;
        </h2>
      </div>
      <hr />
      <div>
        <div
          class="level"
        >
          <div
            class="level-left"
          >
            <div
              class="level-item"
            >
              <span
                class="icon color-success"
                title="Approved"
              >
                <svg
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
                  />
                  <polyline
                    points="22 4 12 14.01 9 11.01"
                  />
                </svg>
              </span>
            </div>
            <div
              class="level-item"
            >
              <strong
                data-testid="reviewer-login"
              >
                &lt;mock-value-for-field-"login"&gt;
              </strong>
            </div>
            <div
              class="level-item"
              data-testid="review-rule-name"
            >
              &lt;mock-value-for-field-"name"&gt;
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
